ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -612,10 +611,10 @@ mkApps kernel Tracers {..} mkCodecs ByteLimits {..} genChainSyncTimeout lopBucke
       bracketWithPrivateRegistry
         (chainSyncHeaderServerFollower
            (getChainDB kernel)
-           ( case isPipeliningEnabled version of
-              ReceivingTentativeBlocks    -> ChainDB.TentativeChain
-              NotReceivingTentativeBlocks -> ChainDB.SelectedChain
-           )
+           ChainDB.TentativeChain
```

Comment:
We do not want to remove the argument from `chainSyncHeaderServerFollower`.

So please remove this comment.

Reply:
On some level, I think this PR already goes too far, removing some logic that is necessary to be able to run a net that doesn't use block diffusion pipelining. `mainnet` does, but these Consensus are supposed to be more general than that.

For `ouroboros-consensus-diffusion`, I could agree with it, since that is glue between Consensus and the Diffusion Layer, which has itself removed the ability to distinguish. However, a separate net might have a default of no pipelining, which this code would then be insufficient for.

The changes I'm already seeing in `ouroboros-consensus` I disagree with -- just because the Diffusion Layer no longer considers pipelining support (it doesn't really assume one way or the other) doesn't mean the Consensus Layer should only support one option.

So in other words, I think the various "is-pipelining-enabled" `Bool` isomorphs should have been pulled out as a new config parameter instead of collapsed to the value used by `mainnet`.

Reply:
That said, all of these changes are quite small, so reintroducing this ability after the fact won't be significant tech debt.

So getting the release out probably takes priority over undoing these changes.

Reply:
Nick, I took the path of least resistance when proposing these adaptations since I'm not privy to the design intent and the issue this caused.; however, now I do see your point and I've refactored to expose pipelining support as a configuration option at the consensus layer since the diffusion layer is now agnostic to it.